web/src/compose_recipient.ts

```diff
@@ -207,20 +225,23 @@ export function on_compose_select_recipient_update() {
     if (curr_message_type === "stream") {
         // Update stream name in the recipient box.
         const stream_id = compose_state.stream_id();
+        assert(stream_id !== undefined);
         update_recipient_label(stream_id);
     }
 
     update_on_recipient_change();
 }
 
-export function possibly_update_stream_name_in_compose(stream_id) {
+export function possibly_update_stream_name_in_compose(stream_id: number): void {
     if (compose_state.selected_recipient_id === stream_id) {
         on_compose_select_recipient_update();
     }
 }
 
-function item_click_callback(event, dropdown) {
-    let recipient_id = $(event.currentTarget).attr("data-unique-id");
+function item_click_callback(event: JQuery.ClickEvent, dropdown: Instance): void {
+    const recipient_id_str = $(event.currentTarget).attr("data-unique-id");
+    assert(recipient_id_str !== undefined);
+    let recipient_id: string | number = recipient_id_str;
```

Comment:
I see what this is doing; it feels very awkward but I'm not sure I have a better theory.